blockpoller/poller.go

```diff
@@ -98,62 +119,178 @@ func (p *BlockPoller) run(resolvedStartBlock bstream.BlockRef) (err error) {
 	}
 }
 
-func (p *BlockPoller) processBlock(currentState *cursor, blkNum uint64) (uint64, error) {
-	if blkNum < p.forkDB.LIBNum() {
-		panic(fmt.Errorf("unexpected error block %d is below the current LIB num %d. There should be no re-org above the current LIB num", blkNum, p.forkDB.LIBNum()))
+func (p *BlockPoller) processBlock(currentState *cursor, block *pbbstream.Block) (uint64, *string, error) {
+	p.logger.Info("processing block", zap.Stringer("block", block.AsRef()), zap.Uint64("lib_num", block.LibNum))
+	if block.Number < p.forkDB.LIBNum() {
+		panic(fmt.Errorf("unexpected error block %d is below the current LIB num %d. There should be no re-org above the current LIB num", block.Number, p.forkDB.LIBNum()))
```

Comment:
Do we really need to panic here? What if we return simply an `error` here? Or terminating right away the block poller is it implements shutter?